internal/cli/kraft/lib/add/add.go

```diff
@@ -0,0 +1,126 @@
+package add
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/spf13/cobra"
+	"kraftkit.sh/cmdfactory"
+	"kraftkit.sh/pack"
+	"kraftkit.sh/packmanager"
+	"kraftkit.sh/unikraft"
+	"kraftkit.sh/unikraft/app"
+	"kraftkit.sh/unikraft/lib"
+)
+
+type AddOptions struct {
+	Workdir    string `long:"workdir" short:"w" usage:"workdir to add the package to"`
+	Kraftfile  string `long:"kraftfile" short:"K" usage:"Set an alternative path of the Kraftfile"`
+	FromServer bool   `long:"from-server" usage:"Force package manager to use manifests of the server"`
+}
+
+// Add adds a Unikraft library to the project directory and updates the Kraftfile
+func Add(ctx context.Context, opts *AddOptions, args ...string) error {
+	if opts == nil {
+		opts = &AddOptions{}
+	}
+
+	return opts.Run(ctx, args)
+}
+
+func NewCmd() *cobra.Command {
+	cmd, err := cmdfactory.New(&AddOptions{}, cobra.Command{
+		Short: "Add unikraft library to the project directory",
+		Use:   "add [FLAGS] [PACKAGE|DIR]",
+		Args:  cmdfactory.MinimumArgs(1, "package Name is not specified to add to the project"),
+		Annotations: map[string]string{
+			cmdfactory.AnnotationHelpGroup: "lib",
+		},
+	})
+	if err != nil {
+		panic(err)
+	}
+
+	return cmd
+}
+
+func (opts *AddOptions) Pre(cmd *cobra.Command, _ []string) error {
+	ctx, err := packmanager.WithDefaultUmbrellaManagerInContext(cmd.Context())
+	if err != nil {
+		return err
+	}
+	cmd.SetContext(ctx)
+	return nil
+}
+
+func (opts *AddOptions) Run(ctx context.Context, args []string) error {
+	var workdir string
+	var err error
+
+	if len(args) > 1 {
+		workdir = args[1]
+	} else {
+		workdir = opts.Workdir
+	}
+
+	if workdir == "." || workdir == "" {
+		workdir, err = os.Getwd()
+	}
+
+	if err != nil {
+		return err
+	}
+
+	nameAndVersion := strings.Split(strings.ToLower(args[0]), ":")
+	var version string
+	if len(nameAndVersion) > 1 {
+		version = nameAndVersion[1]
+	}
+
+	packs, err := packmanager.G(ctx).Catalog(ctx,
+		packmanager.WithName(nameAndVersion[0]),
+		packmanager.WithTypes(unikraft.ComponentTypeLib),
+		packmanager.WithVersion(version),
+		packmanager.WithUpdate(opts.FromServer),
+	)
+	if err != nil {
+		return err
+	}
+
+	if len(packs) != 1 {
+		return fmt.Errorf("specified package not found")
+	}
```

Comment:
Also check for greater than 1 option and create an error that outputs the list of options as an error.

Reply:
Done 👍🏽